crates/modelardb_common/src/storage.rs

```diff
@@ -0,0 +1,222 @@
+pub fn create_apache_arrow_writer<W: Write + Send>(
+    writer: W,
+    schema: SchemaRef,
+    sorting_columns: Option<Vec<SortingColumn>>,
+) -> Result<ArrowWriter<W>, ParquetError> {
+    let props = WriterProperties::builder()
+        .set_encoding(Encoding::PLAIN)
+        .set_compression(Compression::ZSTD(ZstdLevel::default()))
+        .set_dictionary_enabled(false)
+        .set_statistics_enabled(EnabledStatistics::None)
+        .set_bloom_filter_enabled(false)
+        .set_sorting_columns(sorting_columns)
+        .build();
+
+    let writer = ArrowWriter::try_new(writer, schema, Some(props))?;
+    Ok(writer)
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use std::path::PathBuf;
+    use std::sync::Arc;
+
+    use datafusion::arrow::datatypes::{Field, Schema};
+    use tempfile::{self, TempDir};
+
+    use crate::test;
+
+    // Tests for writing and reading Apache Parquet files.
+    #[test]
+    fn test_write_batch_to_apache_parquet_file() {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let batch = test::compressed_segments_record_batch();
+
+        let apache_parquet_path = temp_dir.path().join("test.parquet");
+        write_batch_to_apache_parquet_file(&batch, apache_parquet_path.as_path(), None).unwrap();
+
+        assert!(apache_parquet_path.exists());
+    }
+
+    #[test]
+    fn test_write_empty_batch_to_apache_parquet_file() {
+        let fields: Vec<Field> = vec![];
+        let schema = Schema::new(fields);
+        let batch = RecordBatch::new_empty(Arc::new(schema));
+
+        let temp_dir = tempfile::tempdir().unwrap();
+        let apache_parquet_path = temp_dir.path().join("empty.parquet");
+        write_batch_to_apache_parquet_file(&batch, apache_parquet_path.as_path(), None).unwrap();
+
+        assert!(apache_parquet_path.exists());
+    }
+
+    #[test]
+    fn test_write_batch_to_file_with_invalid_extension() {
+        write_to_file_and_assert_failed("test.txt".to_owned());
+    }
+
+    #[test]
+    fn test_write_batch_to_file_with_no_extension() {
+        write_to_file_and_assert_failed("test".to_owned());
+    }
+
+    fn write_to_file_and_assert_failed(file_name: String) {
+        let temp_dir = tempfile::tempdir().unwrap();
+        let batch = test::compressed_segments_record_batch();
+
+        let apache_parquet_path = temp_dir.path().join(file_name);
+        let result =
+            write_batch_to_apache_parquet_file(&batch, apache_parquet_path.as_path(), None);
+
+        assert!(result.is_err());
+        assert!(!apache_parquet_path.exists());
+    }
+
+    #[tokio::test]
+    async fn test_read_entire_apache_parquet_file() {
+        let file_name = "test.parquet".to_owned();
+        let (_temp_dir, path, batch) = create_apache_parquet_file_in_temp_dir(file_name);
+
+        let result = read_batch_from_apache_parquet_file(path.as_path()).await;
+
+        assert!(result.is_ok());
+        assert_eq!(batch, result.unwrap());
```

Comment:
Maybe verify that the correct number of rows where read from the Apache Parquet file to ensure the PR fixes the issue.

Reply:
Fixed.